### PR TITLE
Use terraform sdk v2 and support importing resources

### PR DESCRIPTION
**Pull request by cristicalin**
This PR updates the provider to use the terraform v2 sdk and implements the ability to use `terraform import` to import pre-created resources from existing phpipam installations.

Fixes: https://github.com/lord-kyron/terraform-provider-phpipam/issues/49

### CODE REVIEW

**Comment by cristicalin** [comment]
@lord-kyron could you take a look at this proposed change?

File: plugin/providers/phpipam/data_source_phpipam_address.go [hunk 1]
```diff
@@ -53,7 +54,19 @@ func dataSourcePHPIPAMAddressRead(d *schema.ResourceData, meta interface{}) erro
...
+    if len(id) > 0 {
+      address_id, err := strconv.Atoi(id)
+      if err != nil {
+        return err
+      }
+		  out[0], err = c.GetAddressByID(address_id)
+		  if err != nil {
+			  return err
+		  }
+    } else {
```
Comment (pavel-z1) [inline plugin/providers/phpipam/data_source_phpipam_address.go:67]: I don't get the point of this change. Can you please add a comment why this check is needed?

File: plugin/providers/phpipam/data_source_phpipam_subnet.go [hunk 2]
```diff
@@ -43,7 +44,19 @@ func dataSourcePHPIPAMSubnetRead(d *schema.ResourceData, meta interface{}) error
...
+    if len(id) > 0 {
+      subnet_id, err := strconv.Atoi(id)
+      if err != nil {
+        return err
+      }
+		  out[0], err = c.GetSubnetByID(subnet_id)
+		  if err != nil {
+			  return err
+		  }
+    } else {
```
Comment (pavel-z1) [inline plugin/providers/phpipam/data_source_phpipam_subnet.go:57]: I don't get the point of this change. Can you please add a comment why this check is needed?

**Review by pavel-z1 — CHANGES_REQUESTED** [review]
First of all, thank you for the work you have done!

Please delete the directory **pkg**. The binary files will be added to the release assets.

In some files need to fix indent. Please format go files with this command to fix indent:
```
gofmt -w plugin/providers/phpipam/*
```

**Comment by cristicalin** [comment]
Hi @pavel-z1, thanks for the review, I made the modifications you requested and explained why I needed to modify the import logic to ensure we don't destroy imported resources.

**Comment by pavel-z1** [comment]
Hi @cristicalin !
Thank you for updates


Format please these files and I will merge PR
```
# go fmt ./plugin/providers/phpipam/*
plugin/providers/phpipam/data_source_phpipam_address.go
plugin/providers/phpipam/data_source_phpipam_subnet.go
```

**Comment by cristicalin** [comment]
@pavel-z1 I `go fmt`'ed the two files

**Review by pavel-z1 — APPROVED** [review]

**Comment by pavel-z1** [comment]
Thank you @cristicalin !
PR merged

**Pull request merged**